packages/fxa-shared/l10n/parseAcceptLanguage.ts

```diff
@@ -0,0 +1,71 @@
+import { negotiateLanguages } from '@fluent/langneg';
+import availableLocales from './supportedLanguages.json';
+import { EN_GB_LOCALES } from './otherLanguages';
+
+/**
+ * Takes an acceptLanguage value (assumed to come from an http header) and returns
+ * a set of valid locales in order of precendence.
+ * @param acceptLanguage - Http header accept-language value
+ * @param supportedLanguages - List of supported language codes
+ * @returns A list of associated supported locales. If there are no matches for the given
+ *          accept langauge, then the default locle, en, will be returned.
+ *
+ */
+export function parseAcceptLanguage(
```

Comment:
After using this code in more places, a few tests started failing. Further investigation revealed that q-values weren't being taken into account. I also felt the code could be improved, so I took the liberty updating it and adding more test coverage. Both tests on parseAcceptLangauge and determineLocale validate this function's logic.